src/components/Proposal/View/AuthorDetails.tsx

```diff
@@ -0,0 +1,153 @@
+import { useMemo } from 'react'
+import { useIntl } from 'react-intl'
+
+import { ProjectStatus } from '../../../entities/Grant/types'
+import { ProposalType } from '../../../entities/Proposal/types'
+import { CURRENCY_FORMAT_OPTIONS } from '../../../helpers'
+import useFormatMessage from '../../../hooks/useFormatMessage'
+import useGovernanceProfile from '../../../hooks/useGovernanceProfile'
+import useProposals from '../../../hooks/useProposals'
+import useVestings from '../../../hooks/useVestings'
+import useVotingStats from '../../../hooks/useVotingStats'
+import locations from '../../../utils/locations'
+import InvertedButton from '../../Common/InvertedButton'
+import Link from '../../Common/Typography/Link'
+import Username from '../../Common/Username'
+import ValidatedProfileCheck from '../../User/ValidatedProfileCheck'
+
+import './AuthorDetails.css'
+import AuthorDetailsStat from './AuthorDetailsStat'
+import Section from './Section'
+
+interface Props {
+  address: string
+}
+
+export default function AuthorDetails({ address }: Props) {
+  const t = useFormatMessage()
+  const { profile, isLoadingGovernanceProfile } = useGovernanceProfile(address)
+  const { participationTotal } = useVotingStats(address)
+  const { proposals: grants } = useProposals({ user: address, type: ProposalType.Grant })
+  const intl = useIntl()
+  const hasPreviouslySubmittedGrants = !!grants && grants?.total > 0
+
+  const { data: vestings } = useVestings(hasPreviouslySubmittedGrants)
+  const grantsWithVesting = useMemo(
+    () =>
+      grants?.data.map((grant) => {
+        const vesting = vestings?.find((item) => grant.id === item.proposal_id)
+
+        return {
+          ...grant,
+          vesting_released: vesting?.vesting_released || 0,
+          vesting_status: vesting?.vesting_status,
+        }
+      }),
+    [vestings, grants?.data]
+  )
+  const fundsReleased = useMemo(
+    () => grantsWithVesting?.reduce((total, grant) => total + grant.vesting_released, 0),
```

Comment:
This memos could be refactored into one so we traverse the list just once

```typescript
const {
  fundsReleased,
  finishedProjectsLength,
  revokedProjectsLength,
  ongoingProjectsLength,
  pausedProjectsLength,
} = useMemo(() => {
  let fundsReleased = 0;
  let finishedProjectsLength = 0;
  let revokedProjectsLength = 0;
  let ongoingProjectsLength = 0;
  let pausedProjectsLength = 0;

  if (grantsWithVesting) {
    grantsWithVesting.forEach((grant) => {
      fundsReleased += grant.vesting_released;

      switch (grant.vesting_status) {
        case ProjectStatus.Finished:
          finishedProjectsLength += 1;
          break;
        case ProjectStatus.Revoked:
          revokedProjectsLength += 1;
          break;
        case ProjectStatus.InProgress:
          ongoingProjectsLength += 1;
          break;
        case ProjectStatus.Paused:
          pausedProjectsLength += 1;
          break;
        default:
          // Handle default case if needed
          break;
      }
    });
  }

  return {
    fundsReleased,
    finishedProjectsLength,
    revokedProjectsLength,
    ongoingProjectsLength,
    pausedProjectsLength
  };
}, [grantsWithVesting]);
```

Reply:
Updated! Check it out!